Adds support for fixed type.

mfranberg opened the pull request:
* Adds a unit test for all currently supported avro types.
* Makes bytes and fixed consistent with hex strings.
* Fixes incorrect casting for byte and enum types.

idreeskhan commented on ratatool-common/src/test/resources/big_sampler_avro_test_schema.avsc:
```diff
@@ -0,0 +1,20 @@
+{
+    "type": "record",
+    "name": "BigSamplerAvroTestRecord",
```
Can just rename to `FlatAvroSchema` or something that better describes the purpose. Any reason to not just use the `RequiredNestedRecord` in the other one?

mfranberg replied:
Good idea I'll use the required ones!

idreeskhan commented on ratatool-sampling/src/main/scala/com/spotify/ratatool/samplers/BigSamplerAvro.scala:
```diff
@@ -60,9 +62,10 @@ private[samplers] object BigSamplerAvro {
             hasher)
-        case Type.ENUM => hasher.putString(v.asInstanceOf[Enum[_]].name, BigSampler.utf8Charset)
+        case Type.ENUM =>
+          hasher.putString(v.asInstanceOf[EnumSymbol].toString, BigSampler.utf8Charset)
```
For my own learning, what is the difference here? Avoids a casting error?

mfranberg replied:
The problem was that you got an EnumSymbol not an Enum and the program failed with an exception. Same happened for Bytes, where I got a ByteBuffer instead of Array[Byte]. Did this work before? :-)

idreeskhan commented on ratatool-sampling/src/test/scala/com/spotify/ratatool/samplers/BigSamplerAvroTest.scala:
```diff
@@ -0,0 +1,94 @@
+import org.apache.avro.generic.{GenericData, GenericRecordBuilder}
+import org.scalatest.{FlatSpec, Matchers}
+
+class BigSamplerAvroTest extends FlatSpec with Matchers {
```
We have some property based tests for hasher already defined in here https://github.com/spotify/ratatool/blob/master/ratatool-sampling/src/test/scala/com/spotify/ratatool/samplers/BigSamplerTest.scala#L36

I think it makes sense to consolidate those into the same file, and keep the JobTests in a separate file.

idreeskhan replied:
Also another note outside scope of this PR, would be nice to rethink some of these as property based tests as well 😉 But not a blocker, up to you

mfranberg replied:
Fixed!

I agree it would be nice to rethink some of these tests! :-)

mfranberg commented:
@idreeskhan Moved the tests into the property based testing file. Added tests for generic records to make sure we get the casting right. As a bonus I refactored the BigSamplerAvro by extracting hashing of the primitive types into its own function.

idreeskhan commented:
This is great, LGTM, I can merge when you are ready @mfranberg

mfranberg commented:
Squashed history, feel free to merge at your convenience!
